src/jwt.rs

```diff
@@ -25,23 +20,20 @@ impl JWT {
     ///
     /// ```
     /// let token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzdWIiOiJsc2t5d2Fsa2VyIiwiaWF0IjoyMzMzNjY0MDB9.k-tTF2CIZ-vu6-syRnCw3Zlc4jwfBCXAQRAyk0mtmso";
-    /// let result = jwtdecode::jwt::JWT::new(token).unwrap();
+    /// let result = jwtdecode::jwt::JWT::new(token.to_string()).unwrap();
     /// assert_eq!(result.body.get("sub").unwrap(), "lskywalker");
     /// ```
-    pub fn new(s: &str) -> JWTResult {
-        let input_string = String::from(s);
+    pub fn new(input_string: String) -> JWTResult {
```

Comment:
So is `String` more common to be a parameter?

Reply:
It's about ownership.  If the function only needs the string temporarily, then it should borrow it with `&str`.  If the function needs to keep it afterwards then it should be given an owned value in the first place with `String`.

Basically make the caller copy, if necessary, rather than copying in the callee, because the caller has more overall context.  Your code is a good example where the caller could give us ownership of the string _without_ making a copy, and thus we've been able to be slightly more efficient (yay).

One reason _not_ to do this is with library public APIs.  Here you should probably stick to what makes conceptual sense because you want to (hopefully) keep the same API across multiple implementations.  So here you should think about "_conceptually_ should this function keep the argument afterwards, or should it just borrow the value temporarily".  Of course, then hopefully the implementation matches that intuition...
Also generally, borrows are more flexible for the caller, particularly for data buffer types.  I can produce a reference-to-a-string (&str) pointing inside heaps of buffer-ish types.  Only one of those types is actually a `String`.